include/mesh/mesh_tools.h

```diff
@@ -129,6 +129,13 @@ std::unordered_set<dof_id_type> find_boundary_nodes(const MeshBase & mesh);
  */
 std::unordered_set<dof_id_type> find_block_boundary_nodes(const MeshBase & mesh);
 
+/**
+ * Returns a std::multimap for all block boundary nodes, listing all subdomain id pairs
+ * for each block boundary the node is on
+ */
+std::map<dof_id_type, std::set<std::pair<subdomain_id_type, subdomain_id_type>>>
+build_block_boundary_node_map(const MeshBase & mesh);
```

Comment:
Hmm... I kind of wish we hadn't called the original routine "find_block" since block is kind of an Exodus term... I'm not sure without looking at it further if we define what "block" means somewhere.

Reply:
I can rename the new one to subdomain

Reply:
Sounds good, thanks!